### ISSUE TITLE
Compiler error when using as a dep

### ISSUE THREAD

**Issue by utx0** [jonhoo/streamunordered#8]
Started to get the following compiler error, and cant for the life of me work out why? 

```
error[E0310]: the parameter type `S` may not live long enough
  --> /home/utx0/.cargo/registry/src/index.crates.io-6f17d22bba15001f/streamunordered-0.5.3/src/task.rs:80:9
   |
80 |         waker_ref(this)
   |         ^^^^^^^^^^^^^^^
   |         |
   |         the parameter type `S` must be valid for the static lifetime...
   |         ...so that the type `S` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound
   |
77 | impl<S: 'static> Task<S> {
   |       +++++++++
```

**Comment by utx0** [comment]
```
❯ rustc -V
rustc 1.81.0 (eeb90cda1 2024-09-04) (Fedora 1.81.0-1.fc40)
```

**Comment by utx0** [comment]
This PR fixed my current issue; https://github.com/jonhoo/streamunordered/pull/9

**Comment by dariusc93** [comment]
The issue is because `futures-util` 0.3.31 added a static lifetime to `wake_ref`.

**Comment by jonhoo** [comment]
Ah, interesting, yes, this is https://github.com/rust-lang/futures-rs/pull/2830, which is a breaking change but was allowed since it was a soundness fix (as per conventions around semver). I'll do the same for `streamunordered` :+1:

**Comment by jonhoo** [comment]
Released in 0.5.4 :tada:
